backend/app/modules/sparql/SparQLActor.scala

```diff
@@ -65,63 +72,75 @@ class SparQLActor(ws: WSClient, serverUrl: String,
     */
   override def receive: Receive = {
     case _: FetchEventClasses => fetchEventClasses()
-    case _: FetchAllGeoJSON => fetchAllGeoJSON()
-    case message: InsertEvent => insertEvent(message)
+    case parameters: CacheParameters => fetchCacheContent(parameters)
+    case event: OntologyEvent => insertEvent(event)
     case _ => throw new InvalidParameterException()
   }
 
   /**
-    * Fetches all geoJSON strings in the fuseki server's ontology
+    * Fetches all event classes of the fuseki server's ontology
+    * Example: Deforestation, Urbanization
     */
-  def fetchAllGeoJSON(): Unit = {
-    val geoJsonRequest = envePrefix + rdfsPrefix +
-      "SELECT ?object WHERE {?subject enve:geoJson ?object}"
+  def fetchEventClasses(): Unit = {
+    val eventClassesRequest = envePrefix + rdfsPrefix +
+      "SELECT ?eventClassName WHERE {?eventClassName rdfs:subClassOf enve:Event}"
 
-    def parseGeoJson(response: WSResponse): Array[String] = {
+    def parseEventClasses(response: WSResponse): Array[String] = {
       val resJson = (response.json \ "results" \ "bindings").as[Array[JsObject]]
-      resJson.map(x => (x \ "object" \ "value").as[String])
+      resJson.map(x => (x \ "eventClassName" \ "value").as[String].split("#")(1))
     }
 
-    sendResponse(executeQuery(geoJsonRequest), parseGeoJson)
+    sendResponse(executeQuery(eventClassesRequest), parseEventClasses)
   }
 
+
   /**
-    * Fetches all event classes of the fuseki server's ontology
-    * Example: Deforestation, Urbanization
+    * Fetches, if they exist, cached results corresponding to the specified parameters
```

Comment:
I would need a description of the response either in the comment or with a case class (if you could add it to sendResponse for that request)